File: app/src/main/java/ru/p3tr0vich/fuel/FragmentInterface.java
package ru.p3tr0vich.fuel;

import android.support.annotation.Nullable;
import android.support.annotation.StringRes;

import ru.p3tr0vich.fuel.factories.FragmentFactory;

public interface FragmentInterface {

    @FragmentFactory.Ids.Id
    int getFragmentId();

    @SuppressWarnings("unused")
    @StringRes
    int getTitleId();

    @Nullable
    String getTitle();

    @SuppressWarnings("unused")
    @StringRes
    int getSubtitleId();

    @Nullable
    String getSubtitle();

    interface OnFragmentChangeListener {
        void onFragmentChange(FragmentInterface fragment);
    }

    /**
     * Обрабатывает нажатие кнопки "назад".
     *
     * @return Истина, если нажатие было обработано во фрагменте,
     * ложь, если нажатие нужно обработать в активности.
     * По умолчанию ложь.
     * @see android.app.Activity#onBackPressed
     */
    boolean onBackPressed();
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/fragments/FragmentInterface.kt
package ru.p3tr0vich.fuel.fragments

import android.support.annotation.StringRes

import ru.p3tr0vich.fuel.factories.FragmentFactory

interface FragmentInterface {

    @get:FragmentFactory.Ids.Id
    val fragmentId: Int

    @get:StringRes
    val titleId: Int?

    val title: String?

    @get:StringRes
    val subtitleId: Int

    val subtitle: String?

    interface OnFragmentChangeListener {
        fun onFragmentChange(fragment: FragmentInterface)
    }

    /**
     * Обрабатывает нажатие кнопки "назад".
     *
     * @return Истина, если нажатие было обработано во фрагменте,
     * ложь, если нажатие нужно обработать в активности.
     * По умолчанию ложь.
     * @see android.app.Activity.onBackPressed
     */
    fun onBackPressed(): Boolean
}